packages/libuv/src-unix-procfs-exepath.c.patch

```diff
@@ -0,0 +1,22 @@
+diff -u -r ../libuv-v1.48.0/src/unix/procfs-exepath.c ./src/unix/procfs-exepath.c
+--- ../libuv-v1.48.0/src/unix/procfs-exepath.c	2024-02-07 20:20:07.000000000 +0000
++++ ./src/unix/procfs-exepath.c	2024-07-02 10:04:51.745846344 +0000
+@@ -26,6 +26,18 @@
+ #include <unistd.h>
+ 
+ int uv_exepath(char* buffer, size_t* size) {
++#ifdef __ANDROID__
++  char* termux_self_exe = getenv("TERMUX_EXEC__PROC_SELF_EXE");
++  if (termux_self_exe) {
++    size_t n = strlen(termux_self_exe);
++    if (*size > n) {
++      memcpy(buffer, termux_self_exe, n);
++      buffer[n] = '\0';
++      *size = n;
++      return 0;
++    }
```

Comment:
An additional enhancement would be to handle `else` case as well, by printing an error and exiting. Seems like libuv uses a buffer of length 500 per default though, so has to be a pretty long path to reach this

Reply:
> but libuv doesn't check the return code when it runs uv_exepath itself internally ([[1]](https://github.com/libuv/libuv/blob/fedfa9893e09dfe92522ec9f4d5a4c3c343484dc/docs/code/proc-streams/main.c#L21), [[2]](https://github.com/libuv/libuv/blob/fedfa9893e09dfe92522ec9f4d5a4c3c343484dc/docs/code/multi-echo-server/main.c#L53)), so it wouldn't catch the issue, which is why I suggested to exit instead.

Those two samples are "just" code samples not checking error return values, not real usage inside the library itself or a recommendation for production. For instance neovim do check the return value.

But indeed, it might be symptomatic that callers don't check the return value if the sample code doesn't, and the current implementation doesn't fail - it just truncates silently instead?

```c
int uv_exepath(char* buffer, size_t* size) {
  ssize_t n;

  if (buffer == NULL || size == NULL || *size == 0)
    return UV_EINVAL;

  n = *size - 1;
  if (n > 0)
    n = readlink("/proc/self/exe", buffer, n);

  if (n == -1)
    return UV__ERR(errno);

  buffer[n] = '\0';
  *size = n;

  return 0;
}
```

So perhaps we should just truncate as well instead, to keep the current behavior of libuv when using the `/proc/self/exe` path (and as you say, errors resulting from that should be clear).

Reply:
Right, then current proposal seems reasonable